annotation/tests/test_distribution.py

```diff
@@ -1277,3 +1291,278 @@ def test_get_page_number_combinations(
     expected: List[int],
 ) -> None:
     assert get_page_number_combinations(file_pages, user_pages) == expected
+
+
+@pytest.mark.parametrize(
+    (
+        "split_multipage_doc_setup",
+        "validation_type",
+        "annotators",
+        "annotation_tasks",
+        "expected_output",
+    ),
+    (
+        (
+            "",
+            ValidationSchema.cross,
+            [{"user_id": "0"}, {"user_id": "1"}],
+            [{"user_id": "0"}, {"user_id": "1"}],
+            [],
+        ),
+        (
+            "1",
+            ValidationSchema.hierarchical,
+            [{"user_id": "0"}, {"user_id": "1"}],
+            [{"user_id": "1"}],
+            [{"user_id": "2"}],
+        ),
+    ),
+)
+def test_choose_validators_users(
+    split_multipage_doc_setup: str,
+    validation_type: ValidationSchema,
+    annotators: List[DistributionUser],
+    annotation_tasks: List[Task],
+    expected_output: List[DistributionUser],
+):
+    validators = [{"user_id": "2"}]
+    with patch(
+        "annotation.distribution.main.SPLIT_MULTIPAGE_DOC",
+        split_multipage_doc_setup,
+    ):
+        output = choose_validators_users(
+            validation_type, annotators, validators, annotation_tasks
+        )
+        assert output == expected_output
+
+
+@pytest.mark.parametrize(
+    ("users_id", "example_db", "expected_output"),
+    (
+        (
+            [UUID("12345678-1234-5678-1234-567812345678")],
+            (UUID("12345678-1234-5678-1234-567812345678"),),
+            [User(user_id=UUID("12345678-1234-5678-1234-567812345678"))],
+        ),
+        (
+            [
+                UUID("12345678-1234-5678-1234-567812345678"),
+                UUID("00d37dc4-b7da-45e5-bcb5-c9b82965351c"),
+            ],
+            (
+                UUID("12345678-1234-5678-1234-567812345678"),
+                UUID("00d37dc4-b7da-45e5-bcb5-c9b82965351c"),
+            ),
+            [
+                User(user_id=UUID("12345678-1234-5678-1234-567812345678")),
+                User(user_id=UUID("00d37dc4-b7da-45e5-bcb5-c9b82965351c")),
+            ],
+        ),
+    ),
+)
+def test_prepare_users(
+    users_id: List[UUID], example_db: Tuple[UUID], expected_output: List[User]
+):
+    db = Mock()
+
+    with patch(
+        "annotation.distribution.main.read_user",
+        side_effect=lambda db, user_id: User(user_id=user_id)
+        if user_id in example_db
```

Comment:
in this test you dont have any case where user_id is not present in example_db. Please adjust test or better add edge cases

Reply:
Im sorry about that. My intention was to cover when user_id is not in example_db with the second test case. I guess in the meantime I copy-pasted the lists from above.